be/src/exec/vectorized/schema_scanner/schema_columns_scanner.cpp

```diff
@@ -197,7 +200,12 @@ Status SchemaColumnsScanner::fill_chunk(ChunkPtr* chunk) {
             // TABLE_SCHEMA
             {
                 ColumnPtr column = (*chunk)->get_column_by_slot_id(2);
-                std::string db_name = SchemaHelper::extract_db_name(_db_result.dbs[_db_index - 1]);
+                std::string db_name;
+                if (_param->without_db_table) {
+                    db_name = _desc_result.columns[_column_index].columnDesc.dbName;
```

Comment:
could we set the dbname once?

Reply:
This is query optimization for "select xxx from information_schema.columns limit 10;", which only call describle_table interface once, so it is different from the previous logic of taking the dbname, so it is necessary to judge if else to get the dbname value